step-framework-parent/step-framework-collections/src/main/java/step/core/accessors/AbstractAccessor.java

```diff
@@ -123,6 +132,9 @@ public Stream<T> stream() {
 	@Override
 	public void remove(ObjectId id) {
 		collectionDriver.remove(byId(id));
+		if (versionedCollectionDriver != null) {

```

Comment:
Deleting an entity would also delete its history which makes rollback impossible. It might be preferable to mark the version as "to be deleted" and delete them asynchronously. One could implement a service to retrieved recently deleted entities and offer their restoration